### ISSUE TITLE
Biocommons does not check duplicated bases after 10 bases

### ISSUE THREAD

**Issue by EmmaTudini** [SACGF/variantgrid#1096]
Found during testing of https://app.zenhub.com/workspaces/everything-space-5bb3158c4b5806bc2beae448/issues/gh/sacgf/variantgrid/857

I think it should check even if it is over 10 bases. Not sure when the limit was changed, in prod it seems to add bases in up to 25 or so bases?

E.g.
According to https://test.shariant.org.au/classification/hgvs_resolution_tool?genome_build=GRCh37
NM_022081.5(HPS4):c.949_958dupGCTTGTCCAG (bases not blanked)

But 

NM_022081.5(HPS4):c.949_959dupGCTTGTCCAGC - bases blanked and not checked

- Note to check how this resolved on the imported allele popup on the allele page, once biocommons switched over in test.

**Comment by davmlaw** [comment]
I don't know what you mean by checked, and I don't know what you expect by adding.

Test and prod looks the same to me in Biocommons:

| Input | Test | Prod |
| ----------- | ------------ | ------------- |
| NM_022081.5(HPS4):c.949_958dupGCTTGTCCAG | NM_022081.5(HPS4):c.949_958dupGCTTGTCCAG  | NM_022081.5(HPS4):c.949_958dupGCTTGTCCAG |
| NM_022081.5(HPS4):c.949_959dupGCTTGTCCAGC | NM_022081.5(HPS4):c.949_959dup | NM_022081.5(HPS4):c.949_959dup (prod) |

Can you please explicitly list what the differences are in prod/test and what you expect values to be when you say they aren't right? Thanks

**Comment by EmmaTudini** [comment]
@davmlaw No difference between test and prod, but a difference between Pyhgvs and Biocommons, which would be good to resolve. 

With pyhgvs, we actually checked that the bases specified in the dup/del were correct, but with Biocommons, we seem to ignore the bases specified in the input.

**Comment by davmlaw** [comment]
> With pyhgvs, we actually checked that the bases specified in the dup/del were correct

I think PyHGVS doesn't check anything

PyHGVS inserts the reference sequence that you provide without any checking at all, ie  PyHGVS accepts:

```
NM_022081.5(HPS4):c.949_958dupAAAAAAAAAA
NM_022081.5(HPS4):c.949_958dupCCCCCCCCCC
NM_022081.5(HPS4):c.949_958dupGGGGGGGGGG
NM_022081.5(HPS4):c.949_958dupTTTTTTTTTT
```

And inserts them as different reference bases

**Comment by davmlaw** [comment]
Ah, I think I know what you mean about "check duplicated/deleted bases after 10 bases"

I thought you meant check against the reference but what you mean is check against the submitted HGVS after re-generating it.

So at the moment Biocommons replaces them with the reference, which causes a check in the "significant change to HGVS", and when it strips them, you get a difference but only by seeing the bases disappear

So focusing on new behavior: what would you like it to do if the provided reference doesn't match our calculated one?

On imports - raise a warning on search? Fail? Fail imports? Raise a warning saying bases don't match?

**Comment by davmlaw** [comment]
@EmmaTudini are you able to provide expected behavior? Thanks

**Comment by EmmaTudini** [comment]
I would like Biocommons to validate the duplicated or deleted bases - we catch any issues with Pyhgvs as Pyhgvs inserts the bases and we do a comparison between imported and resolved at the end. Biocommons does not insert and therefore the validation step doesn't happen.

**Comment by davmlaw** [comment]
What do you mean by "validate", ie what would you like it to do if the provided reference doesn't match our calculated one?

On imports raise a warning or error? On search?

**Comment by EmmaTudini** [comment]
@davmlaw Were any changes to the biocommons code done for this issue>

**Comment by davmlaw** [comment]
No. I think biocommons has correct behavior here so can just use it as is.
